src/wallet/test/fuzz/coinselection.cpp

```diff
@@ -118,6 +118,17 @@ FUZZ_TARGET(coinselection)
     if (total_balance >= target && subtract_fee_outputs && !HasErrorMsg(result_knapsack)) {
         assert(result_knapsack);
     }
+
+    std::vector<COutput> utxos;
+    std::set<std::shared_ptr<COutput>> utxos_set;
+    CreateCoins(fuzzed_data_provider, utxos, coin_params);
+    for (const auto& utxo : utxos) {
+        utxos_set.insert(std::make_shared<COutput>(utxo));
+    }
+
+    if (result_srd) result_srd->AddInputs(utxos_set, subtract_fee_outputs);
+    if (result_knapsack) result_knapsack->AddInputs(utxos_set, subtract_fee_outputs);
+    if (result_bnb) result_bnb->AddInputs(utxos_set, subtract_fee_outputs);
```

Comment:
what about asserting inputs are actually added and checking the total weight?

Reply:
Sounds good. Just did it.